pandas/_libs/parsers.pyx

```diff
@@ -1192,7 +1192,12 @@ cdef class TextReader:
         # only allow safe casts, eg. with a nan you cannot safely cast to int
         if col_res is not None and col_dtype is not None:
             try:
+                mask = None
+                if col_res.dtype == np.bool_ and na_count > 0:
+                    mask = col_res.view(np.uint8) == na_values[np.uint8]
```

Comment:
i am not sure this is the right place to do this, try in _try_bol

Reply:
I can't put this in _try_bool because the problem is with the line 1198: col_res.astype("float32").
The function astype when we have a vector of type np.bool_ like this:
[False NaN]
Transforms it to this:
[0.0 1.0]
Because the na_value of the type np.bool_ is 255.

Reply:
```
In [4]: np.array([False, np.nan], dtype=object).astype('float32', casting='safe')
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-4-6afb8b0a6a7e> in <module>()
----> 1 np.array([False, np.nan], dtype=object).astype('float32', casting='safe')

TypeError: Cannot cast array from dtype('O') to dtype('float32') according to the rule 'safe'
```
what is the type of ``col-res`` coming into here?